appscale/tools/agents/ec2_agent.py

```diff
@@ -477,6 +582,13 @@ def run_instances(self, count, parameters, security_configured, public_ip_needed
         AppScaleLogger.log("Waiting for your instances to start...")
         public_ips, private_ips, instance_ids = self.describe_instances(
           parameters)
+
+        # If we need a public ip, make sure we actually get one.
+        if public_ip_needed and not self.diff(public_ips, private_ips):
+          time.sleep(self.SLEEP_TIME)
+          now = datetime.datetime.now()
```

Comment:
To make this simpler and prevent future updates here from accidentally forgetting to update `now`, just replace `now` in the if statement and remove the assignments.

Reply:
Just want to make sure I'm understanding correctly, you mean do `while datetime.datetime.now() < end_time:`?

Reply:
Yes, that's right.